lib/signaling/v2/room.js

```diff
@@ -719,6 +733,58 @@ function handleSubscriptions(room) {
   });
 }
 
+function addAVSyncMetricsToLocalTrackStats(trackStats, trackResponse) {
+  const {
+    framesEncoded,
+    packetsSent,
+    totalEncodeTime,
+    totalPacketSendDelay
+  } = trackResponse;
+  const augmentedTrackStats = Object.assign({}, trackStats);
+
+  if (typeof totalEncodeTime === 'number' && typeof framesEncoded === 'number') {
+    const totalEncodeTimeMs = totalEncodeTime * 1000;
+    augmentedTrackStats.avgEncodeDelay = Math.round(framesEncoded <= 0
```

Comment:
@manjeshbhargav, I was thinking that these metrics would calculate the rates only for the duration since last metric was sent. so avgEncodeDelay would be calculated as:
```
deltaEncodeTime = totalEncodeTimeMS - lastTotalEncodeTimeMS;
lastTotalEncodeTimeMS = totalEncodeTimeMS;
deltaFrameEncoded = framesEncoded - lastFramesEncoded;
lastFramesEncoded = framesEncoded;
avgEncodeDelay = deltaFrameEncoded / deltaEncodeTime
``` 

That way those values would have meaningful spikes and delays can be detected by drawing a graph of those delays. If thy are average over the lifetime of peerConnection - I am afraid it wont be noticeable.

Maybe @ceaglest can tell us if these values would be detectable if they are averaged over longer times.

Reply:
@makarandp0 If we want to do this, then it cannot be called as the average. We would need to remove the prefix `avg` from the fields.

Reply:
I don't think an average over all time is going to be very helpful to debug media quality problems. The issue is that the longer the stats are collected the harder it becomes to see momentary spikes or troughs in these metrics.

For context I originally advocated for reporting totals and having the gateway compute the average (total, moving or otherwise).

> @makarandp0 If we want to do this, then it cannot be called as the average. We would need to remove the prefix avg from the fields.

What we intended was a [moving average](https://en.wikipedia.org/wiki/Moving_average) with a window size of two samples. It is an average but over a much smaller data set. If you are concerned that the field naming is not clear we could change the prefix to `movingAvg`, `rollingAvg` or `runningAvg` instead.